chain/default_store.go

```diff
@@ -359,94 +358,19 @@ func (store *DefaultStore) writeTipSetAndState(tsas *TipSetAndState) error {
 	return store.ds.Put(key, val)
 }
 
-// Head returns the current head.
-func (store *DefaultStore) Head() types.TipSet {
+// GetHead returns the current head.
+func (store *DefaultStore) GetHead() types.SortedCidSet {
```

Comment:
What motivates this change? It seems inconvenient and led to lots of callers to change. Is it to hide the head tipset to prevent external modification?

Please document why it's structured this ways so that a future developer who finds it inconvenient doesn't mistakenly change it back without context.

Reply:
@anorth I asked the same question and @phritz left an in depth answer in slack.  I'll tag you there so you get access to it.

Reply:
I'm also confused about this change. It seems unnatural and unnecessary. Could @phritz, @ZenGround0 or @rosalinekarr respond in this PR?

Reply:
Copy-paste from [slack](https://filecoinproject.slack.com/archives/CEHHJNJS3/p1554234370122200?thread_ts=1554213262.096100&cid=CEHHJNJS3):

> there are a lot of considerations for how to define good apis but here are a few generalizations it is hard to go wrong with: 1. push complexity onto the caller side of the api; generally dont provide convenience functions if your callers could. 2. keep your apis as small as possible 3. keep your calls as orthogonal as possible. All three of these principles are in evidence in the new api, which is not perfect. It 1 eliminates convenience methods where a caller can simply make two calls and 2 reduces the number of calls in the api and 3 makes the calls that remain more orthogoal, though not completely so. having apis that follow these principles make the api easier to optimize, easier to understand, and easier to satisfy with a variety of implementations. they limit the blast radius of changes, should you have to make them. martin fowler, one of the gang of four, identifies this philosophy of api design as the minimal interface and articulates some of its benefits here https://martinfowler.com/bliki/MinimalInterface.html.  this is in contrast with sprawling expansive convenience interfaces he calls humane interfaces https://martinfowler.com/bliki/HumaneInterface.html. the systems and distributed systems worlds are strongly oriented around minimal interfaces these days for the reasons articulated above. many web frameworks eg rails are oriented around the latter in the name of developer productivity. I find the maintenance and understandability of narrow interfaces to be far more compelling than saving a few lines when writing a consumer of the interface.